services/src/datasets/listing.rs

```diff
@@ -0,0 +1,98 @@
+use crate::datasets::storage::{AddDataSetProvider, AddMockDataSetProvider, DataSet};
+use crate::error::Result;
+use crate::users::user::UserId;
+use crate::util::user_input::{UserInput, Validated};
+use async_trait::async_trait;
+use geoengine_datatypes::dataset::DataSetId;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct DataSetListing {
+    pub id: DataSetId,
+    pub name: String,
+    pub description: String,
+    pub tags: Vec<String>,
+    pub source_operator: String, // TODO: enum?
```

Comment:
Rather an Operator that serializes to its name.

Reply:
would be the best option. do we need deserialize here? Then i dont know how this would work